src/main/kotlin/no/nav/familie/ef/sak/service/JournalføringService.kt

```diff
@@ -0,0 +1,133 @@
+package no.nav.familie.ef.sak.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import no.nav.familie.ef.sak.api.journalføring.JournalføringRequest
+import no.nav.familie.ef.sak.domene.DokumentBrevkode
+import no.nav.familie.ef.sak.domene.DokumentVariantformat
+import no.nav.familie.ef.sak.integration.JournalpostClient
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.kontrakter.ef.søknad.SøknadBarnetilsyn
+import no.nav.familie.kontrakter.ef.søknad.SøknadOvergangsstønad
+import no.nav.familie.kontrakter.ef.søknad.SøknadSkolepenger
+import no.nav.familie.kontrakter.felles.dokarkiv.*
+import no.nav.familie.kontrakter.felles.journalpost.Dokumentvariant
+import no.nav.familie.kontrakter.felles.journalpost.Journalpost
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.oppgave.Oppgavetype
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.util.*
+
+@Service
+class JournalføringService(private val journalpostClient: JournalpostClient,
+                           private val behandlingService: BehandlingService,
+                           private val oppgaveService: OppgaveService) {
+
+    private val logger = LoggerFactory.getLogger(JournalføringService::class.java)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun hentJournalpost(journalpostId: String): Journalpost {
+        return journalpostClient.hentJournalpost(journalpostId)
+    }
+
+    fun hentDokument(journalpostId: String, dokumentInfoId: String): ByteArray {
+        return journalpostClient.hentDokument(journalpostId, dokumentInfoId)
+    }
+
+    fun fullførJournalpost(journalføringRequest: JournalføringRequest, journalpostId: String): Long {
```

Comment:
@Transactional

Reply:
Men gir det mening her - med alle eksterne parter?

- Oppdaterer Behandling med tilhørende journalpostId
- Oppdaterer intern OppgaveBehandling-kobling

De tre andre tingene som kan gå til h**** er jo http-kall som vi ikke kontrollerer 🤔

Reply:
Det hjelper for oppdateringer av databasen, er mange kall mot databasen. 2 eksterne kall og minimum 4 oppdateringer mot databasen? 
Samtidig lurer jeg på om vi burde prøve på å få denne typen av hendelser som async, dvs vi oppretter en task for de, slik att hvis ett kall mot en extern tjeneste feiler, så prøver tasken på nytt. Har vi peiling på hvordan de andre systemene håndter idempotens? 
Du og jeg har ju dårlig erfaring fra disse tingene siden tidligere :P 

Alternativt legge til `@Retry`? 
@bent-lorentzen input?

Reply:
Oppdateringer har nok en "grad av" idempotens, men opprettinger har neppe det.